core/domain/state_domain_test.py

```diff
@@ -637,6 +639,572 @@ def test_validate_state_solicit_answer_details(self):
         init_state = exploration.states[exploration.init_state_name]
         self.assertEqual(init_state.solicit_answer_details, True)
 
+    def test_validate_solution_answer_is_exclusive(self):
+        exploration = self.save_new_valid_exploration('exp_id', 'owner_id')
+
+        # Solution should be set to None as default.
+        self.assertEqual(exploration.init_state.interaction.solution, None)
+
+        hints_list = []
+        solution = {
+            'answer_is_exclusive': False,
+            'correct_answer': 'hello_world!',
+            'explanation': {
+                'content_id': 'solution',
+                'html': 'hello_world is a string'
+            }
+        }
+        hints_list.append({
+            'hint_content': {
+                'content_id': 'hint_1',
+                'html': ''
+            },
+        })
+        exploration.init_state.update_interaction_hints(hints_list)
+        exploration.init_state.update_interaction_solution(solution)
+        exploration.validate()
+
+        solution = {
+            'answer_is_exclusive': 1,
+            'correct_answer': 'hello_world!',
+            'explanation': {
+                'content_id': 'solution',
+                'html': 'hello_world is a string'
+            }
+        }
+
+        exploration.init_state.update_interaction_solution(solution)
+        with self.assertRaisesRegexp(
+            Exception, 'Expected answer_is_exclusive to be bool'):
```

Comment:
Can we add the full log wherever possible?

Reply:
done